### ISSUE TITLE
UiBinder file location in gwt-lib artifacts

### ISSUE THREAD

**Issue by timeu** [tbroyer/gwt-maven-plugin#33]
I guess this is by design but probably should be documented: 

In a `gwt-lib` library one has to put the UiBinder files (i.e. `SomeView.ui.xml`) into the `resources/PACKAGE` folder, so that they are packaged when running `mvn package`.  
If they UIBinder files are in the `src/main/java/PACKAGE` folder, they are **not** packaged.

**Comment by tbroyer** [comment]
Even if you explicitly declare `src/main/java` as a `<resource>` in your POM?

(unfortunately, I have no idea why I have added the `**/*.java` include filter: 4491107261d685bae8758d83a8ae8c8844d993b6)

Maybe we should rather just remove it as `src/main/java` would work OK in a `gwt-app`, and more importantly also in a `gwt-lib` when in a reactor project with `gwt:codeserver` or `gwt:devmode` (as they directly add the source roots to the classpath); and this is also what `source:jar` would do AFACT.

**Comment by timeu** [comment]
I didn't try to declare it explicitly (I removed it when moving from the "official" gwt-maven-module to your's)  but I guess it should work fine if I do (can try it if you like). 

I stumbled over this when I tried to run `mvn package` on a reactor project that consists of a `gwt-lib` library and a sample project that uses the library. I got this error: 

```
[INFO]    Computing all possible rebind results for 'com.github.timeu.gwtlibs.gwasgeneviewer.client.SettingsPanel.Binder'
[INFO]       Rebinding com.github.timeu.gwtlibs.gwasgeneviewer.client.SettingsPanel.Binder
[INFO]          Invoking generator com.google.gwt.uibinder.rebind.UiBinderGenerator
[INFO]             [ERROR] Unable to find resource: com/github/timeu/gwtlibs/gwasgeneviewer/client/SettingsPanel.ui.xml
[INFO]    [ERROR] Errors in 'com/github/timeu/gwtlibs/gwasgeneviewer/client/SettingsPanel.java'
[INFO]       [ERROR] Line 26: Failed to resolve 'com.github.timeu.gwtlibs.gwasgeneviewer.client.SettingsPanel.Binder' via deferred binding
[INFO]    [WARN] For the following type(s), generated source was never committed (did you forget to call commit()?)
[INFO]       [WARN] com.github.timeu.gwtlibs.gwasgeneviewer.client.SettingsPanel_BinderImpl
```

When I move the `SettingsPanel.ui.xml` to the `resources` folder it works fine. 

However as you said, running `gwt:devmode` on the reactor project works fine even when the UIBinder files are in `src/main/java`.

**Comment by timeu** [comment]
There is also another issue when the UiBinder files are in `resources/PACKAGE` when running dev-mode. Changes to the UIBinder won't be picked up when recompiling. This works if they are in `src/main/java`. So I guess either we remove the filter or add `**/*.gwt.xml`.

**Comment by tbroyer** [comment]
> There is also another issue when the UiBinder files are in `resources/PACKAGE` when running dev-mode. Changes to the UIBinder won't be picked up when recompiling. This works if they are in `src/main/java`.

This is a feature ;-)

Resources can be "filtered", so they **have** to be processed by `resources:resources`. This can possibly done automatically by your IDE (IIRC, Eclipse does this)

I think we should just remove the filter.

**Comment by ibaca** [comment]
I get into the same situation after migrating from mojo gwt-maven-plugin, I created this issue https://github.com/tbroyer/gwt-maven-plugin/pull/32 but as @tbroyer said, just adding the resources solves the problem. But, if this problem is common from people coming from mojo plugin, probably removing the filter or adding some warning if ui.xml o gwt.xml are found in sources might be nice.

**Comment by tbroyer** [comment]
Ah right. I hadn't realized at the time that the behavior would be different between `gwt-app` and `gwt-lib`, and between `gwt:compile` and `gwt:codeserver`/`gwt:devmode`. I also think the plugin should not check whether a "conflicting" resource exists when adding `src/main/java`, for the same reason: source roots are added to the classpath of the compiler, devmode and codeserver, where there's no filter (the alternative would be to copy the source roots and add the copy to the classpath, and continuously watch the source roots and sync the changes to the copy; I refuse to do such a thing).

**Comment by tbroyer** [comment]
Fixed by a15aa99b6770e6f672ed041ec58fe8737d6d7006
